### PR TITLE
Reorganized the documents structure 

### PR DESCRIPTION
**Pull request by haoel**
1. Move the reference document into `./doc/reference` directory.
2. Write the `./doc/README.md` to index all of the documents.
3. remove the original `filters.md` and `cookbook/README.md`.
4. rename the original `reference.md` to `filters.md`
5. move all of the images into `doc/imgs/` directory. 
6. rename the file name, change the `_` to `-`

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	- [2. Development Guide](#2-development-guide)
+	- [3. Performance](#3-performance)
+	- [4. Reference Guide](#4-reference-guide)
+		- [4.1 Controllers](#41-controllers)
+			- [4.1.1 System Controllers](#411-system-controllers)
+			- [4.1.2 Business Controllers](#412-business-controllers)
+		- [4.2 Filters](#42-filters)
+
+## 1. Cookbook / How-To Guide
+This is a cook book lists a number of useful and practical examples to use the Easegress for different scenarios.
```
Comment (samutamm) [inline doc/README.md:14]:
```suggestion
This is a cookbook that lists a number of useful and practical examples how to use the Easegress for different scenarios.
```

File: doc/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+### 4.1 Controllers
+
+The Easegress controller is the core entity to control kinds of working. There are two kinds of controllers - system and business. 
+
+The full document please check - [Controller Reference](./reference/controllers.md)
+
+#### 4.1.1 System Controllers
+
+The following controllers are system level controllers,  they are created one and only one instance in every Easegress node, which can't be deleted.
```
Comment (samutamm) [inline doc/README.md:54]:
```suggestion
The following controllers are system level controllers.  One and only one instance of them are created in every Easegress node and they can't be deleted. 
```

File: doc/reference/controllers.md [hunk 3]
```diff
@@ -149,9 +149,11 @@ kafka:
 | ----- | ---------------------------------------------------- | -------------------- | -------- |
 | kafka | [easemonitormetrics.Kafka](#easemonitormetricsKafka) | Kafka related config | Yes      |
 
-### Function
+### FaaSController
 
-TODO (@ben)
+A FaaSController is a business controller for handling Easegress and FaaS products integration purposes.  It abstracts `FaasFunction`, `FaaSStore` and, `FaasProvider`. Currently, we only support `Knative` type `FaaSProvider`. 
+
+The full reference document please check - [FaaS Controller](./faascontroller.md)
```
Comment (samutamm) [inline doc/reference/controllers.md:156]:
```suggestion
For the full reference document please check - [FaaS Controller](./faascontroller.md)
```

**Review by suchen-sci — COMMENTED** [review]
the new structure of doc looks pretty good! 

Is that a good idea to put all images into a single directory? 

I find some file name connect words use "_" and some use "-" for example "api_aggregator.md" and "developer-guide.md". Is that a good idea to make all md file use same pattern?

**Comment by haoel** [comment]
> the new structure of doc looks pretty good!
> 
> Is that a good idea to put all images into a single directory?
> 
> I find some file name connect words use "_" and some use "-" for example "api_aggregator.md" and "developer-guide.md". Is that a good idea to make all md file use same pattern?

those are good ideas, let me work them out!

File: doc/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	- [2. Development Guide](#2-development-guide)
+	- [3. Performance](#3-performance)
+	- [4. Reference Guide](#4-reference-guide)
+		- [4.1 Controllers](#41-controllers)
+			- [4.1.1 System Controllers](#411-system-controllers)
+			- [4.1.2 Business Controllers](#412-business-controllers)
+		- [4.2 Filters](#42-filters)
+
+## 1. Cookbook / How-To Guide
+This is a cookbook that lists a number of useful and practical examples how to use the Easegress for different scenarios.
```
Comment (localvar) [inline doc/README.md:14]:
```suggestion
## 1. Cookbook / How-To Guide

This is a cookbook that lists a number of useful and practical examples on how to use Easegress for different scenarios.
```

File: doc/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+- [Workflow](./cookbook/workflow.md) - An Example to make a workflow for a number of APIs.
+
+
+
+## 2. Development Guide
+
+- [Easegress Roadmap](./Roadmap.md) - The development roadmap of Easegress.
+- [Developer Guide](./developer-guide.md) - A guide help to develop the Easegress.
+
+## 3. Performance
```
Comment (localvar) [inline doc/README.md:40]:
```suggestion
## 2. Development Guide

- [Easegress Roadmap](./Roadmap.md) - The development roadmap of Easegress.
- [Developer Guide](./developer-guide.md) - A guide help to develop the Easegress.

## 3. Performance

```

File: doc/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- [Benchmark](./reference/benchmark.md) - Performance Test Report.
+- [Linux Kernel Tuning](./reference/kernel-tuning.md) - Tuning the Linux Kernel to make the Easegress run faster.
+
+## 4. Reference Guide
+
+### 4.1 Controllers
+
+The Easegress controller is the core entity to control kinds of working. There are two kinds of controllers - system and business. 
+
+The full document please check - [Controller Reference](./reference/controllers.md)
```
Comment (localvar) [inline doc/README.md:50]:
```suggestion
For the full document, please check - [Controller Reference](./reference/controllers.md)
```

File: doc/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+- [MeshController](./reference/controllers.md#meshcontroller) - This is for [EaseMesh](https://github.com/megaease/easemesh) project.
+- [ConsulServiceRegistry](./reference/controllers.md#consulserviceregistry) - supports service discovery for Consul as backend. 
+- [EtcdServiceRegistry](./reference/controllers.md#etcdserviceregistry) - support service discovery for Etcd as backend. 
+- [EurekaServiceRegistry](./reference/controllers.md#eurekaserviceregistry) - supports service discovery for Eureka as backend. 
+- [ZookeeperServiceRegistry](./reference/controllers.md#zookeeperserviceregistry) -  supports service discovery for Zookeeper as backend. 
+- [NacosServiceRegistry](./reference/controllers.md#nacosserviceregistry) - supports service discovery for Nacos as backend.
+- [AutoCertManager](./reference/controllers.md#autocertmanager) - automatically manage HTTPS certificates. 
+
+
+### 4.2 Filters
```
Comment (localvar) [inline doc/README.md:76]:
```suggestion
### 4.2 Filters

```

**Review by suchen-sci — APPROVED** [review]

**Review by localvar — APPROVED** [review]

**Pull request merged**